### DISCUSSION TITLE
Elog1.0 共建计划寻找小伙伴

### DISCUSSION THREAD

**Discussion by LetTTGACO** [LetTTGACO/elog#72] [Announcements]
## 需求描述
[【Elog开发计划】](https://1874.notion.site/Elog1-0-85ca6db36a614253be22bbe5571ec81e)

## 现存 Elog 痛点

- 一对多场景配置复杂。用户想从语雀分别同步到 Hexo（本地）和 Halo 时，需要配置多个配置文件和多个运行命令。
- 同步流程紊乱。流程架构设计历史问题，同步流程不是很清晰。
- 自定义处理复杂。用户想自己处理一些特殊场景时需要用到自定义处理配置复杂且流程不清晰。
- 配置文件不支持 TypeScript。没有 ts 智能补全和校验，对于不太懂编程的用户不友好。
- 不支持按需引入插件。现存 Elog 虽然是 monorepo，但实际在配置和运行时，虽然只安装一个 cli 包，但是会同步引入所有平台的包，即使用不到。发包的时候也是同步升级所有依赖包。总体来说就是假的 monorepo，用起来还是一个整体。

## Elog 1.0 重点功能

- 配置文件直接支持一对多，流程更清晰
- 按需引入需要用到插件包，而不是之前安装一个 cli 包
- Elog 核心只负责统筹流程，文档同步和处理全部交给插件系统去解决
- 插件系统和自定义钩子，让自定义处理更方便
- 全面拥抱 esm，放弃 commonjs

## 技术栈
- 面向平台：Node
- 开发语言：typescript
- monorepo 包管理：rush + pnpm
- 打包模块：全面拥抱 esm，放弃 commonjs

## 目前进度
- 同步流程初版也确立
![image](https://github.com/LetTTGACO/elog/assets/37357188/ecef9ad4-25db-42a9-a52b-5ef767e57bd9)

- 文件夹已创建[旺柴]
<img width="911" alt="image" src="https://github.com/LetTTGACO/elog/assets/37357188/d1493a1a-4585-410a-bdf9-9d584a28d121">

## 欢迎所有感兴趣小伙伴参与，如果你有想法，可加入微信群中一起讨论，让 Elog 变的更好！
![image](https://github.com/LetTTGACO/elog/assets/37357188/7e96f2ed-af40-467f-bf0c-470c7e36d9ef)

**Comment by gylove1994** [comment]
二维码已过期

**Reply by LetTTGACO** [reply]
![image](https://github.com/LetTTGACO/elog/assets/37357188/e0c83705-85dd-42b3-b8de-615cb800d7e6)
👏 欢迎加入

**Reply by LetTTGACO** [reply]
> @LetTTGACO 
> 
> 可以补发下群二维码吗
欢迎加入
![image](https://github.com/user-attachments/assets/0ef127af-11b0-4cca-98df-244a8a8e0ca6)

**Comment by white0dew** [comment]
你好，我想加入，请问怎么加群呢，过期了。 建议你给一个语雀链接，定时把图片放在语雀链接就好

**Reply by white0dew** [reply]
> 你好，我想加入，请问怎么加群呢，过期了。 建议你给一个语雀链接，定时把图片放在语雀链接就好

@LetTTGACO

**Reply by LetTTGACO** [reply]
> 你好，我想加入，请问怎么加群呢，过期了。 建议你给一个语雀链接，定时把图片放在语雀链接就好

麻烦发个邮件给我 me@mail.1874.cool
我拉你进群
